From cddd9de4c63a3c9f54cf0789f83b1af135143b71 Mon Sep 17 00:00:00 2001
From: Lucas Russo <lerwys@gmail.com>
Date: Fri, 4 Mar 2016 10:57:12 -0300
Subject: [PATCH] Problem: zauth_v2 sometimes hangs in self test

This is a backport to the zauth hanging problem
described in commit zeromq/czmq@b6805df78 by hintjens.
---
 src/zauth_v2.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/zauth_v2.c b/src/zauth_v2.c
index c44ffba..1d6a762 100644
--- a/src/zauth_v2.c
+++ b/src/zauth_v2.c
@@ -615,9 +615,15 @@ s_can_connect (zctx_t *ctx, void **server, void **client)
     assert (port_nbr > 0);
     int rc = zsocket_connect (*client, "tcp://127.0.0.1:%d", port_nbr);
     assert (rc == 0);
+    //  Give the connection time to fail if that's the plan
+    zclock_sleep (200);
+
+    //  By default PUSH sockets block if there's no peer
+    zsock_set_sndtimeo (*server, 200);
     zstr_send (*server, "Hello, World");
+
     zpoller_t *poller = zpoller_new (*client, NULL);
-    bool success = (zpoller_wait (poller, 200) == *client);
+    bool success = (zpoller_wait (poller,400) == *client);
     zpoller_destroy (&poller);
     zsocket_destroy (ctx, *client);
     zsocket_destroy (ctx, *server);
-- 
1.8.3.1

